gpflow/likelihoods/base.py

```diff
@@ -294,6 +295,86 @@ def variational_expectations(self, Fmu, Fvar, Y):
     def _variational_expectations(self, Fmu, Fvar, Y):
         raise NotImplementedError
 
+    # @abc.abstractmethod
+    def conditional_sample(self, Fsample) -> tf.Tensor:
+        """ Sample from the likelihood """
+
+
+class ConditionedLikelihood:
+    """
+    This class represents the likelihood of the output conditioned on the latent
+    process(es) function values.
+    """
+
+    def __init__(self, likelihood: Likelihood, Fmean: tf.Tensor, Fvar: tf.Tensor) -> None:
+        """
+        :param likelihood: model likelihood
+        :param Fmean: latent process mean values obtained at some points X.
+        :param Fvar: latent process variance values ontained at some points.
```

Comment:
what is the shape of `Fvar`? Are we only dealing with the marginal variances (i.e. `Fvar` has shape `[N, L]`) or the full covariance (`Fvar` has shape `[L, N, N]`), where `N` is the number of inputs and `L` the number of latent GPs.